site/client/src/main/scala/Ui.scala

```diff
@@ -5,54 +5,49 @@ import IO._
 import actions._
 import model._
 import model.Exercises._
-import utils.{ DomHandler }
+import utils.DomHandler._
 
 object UI {
   def noop: IO[Unit] = io {}
 
   def update(s: State, a: Action): IO[Unit] = a match {
-    case UpdateExercise(method, args) ⇒ toggleCompileButton(s, method)
+    case UpdateExercise(method, args) ⇒ toggleExerciseClass(s, method)
     case CompileExercise(method)      ⇒ startCompilation(s, method)
     case CompilationOk(method)        ⇒ setAsSolved(s, method)
     case CompilationFail(method)      ⇒ setAsErrored(s, method)
     case _                            ⇒ noop
   }
 
-  def toggleCompileButton(s: State, method: String): IO[Unit] = {
+  def toggleExerciseClass(s: State, method: String): IO[Unit] = {
     Exercises.findByMethod(s, method) match {
       case Some(exercise) ⇒ {
         if (exercise.isFilled)
-          enableCompileButton(method)
+          setClassToExercise(method, "filled")
         else
-          disableCompileButton(method)
+          setClassToExercise(method, "unsolved")
       }
       case _ ⇒ noop
     }
   }
 
-  def enableCompileButton(method: String): IO[Unit] = {
-    DomHandler.nodeByMethod(method)
-      .map(s ⇒ s.foreach(
-        (node ⇒ DomHandler.enableCompileButton(node).unsafePerformIO())
-      ))
-  }
+  def setClassToExercise(method: String, style: String): IO[Unit] = findExerciseByMethod(method)
+    .map(_.foreach(setClass(_, style).unsafePerformIO()))
 
```

Comment:
Since `setClass` performs mutations on the DOM it returns an `IO[Unit]` and we need to execute it, I'm not entirely happy with this either but I wanted to reflect the side effects in the type signature.